From: Jory A. Pratt <anarchy@gentoo.org>
Adapted for torbrowser by Poncho <poncho@spahan.ch>

Note: the first occurrence of :space: is [:space:] (with a single pair
of []), where it should be [[:space:]] (with two pairs of []).

This causes the sed command to fail (with the error message quoted in
my original report), driving to failure the whole configure step.

https://bugzilla.mozilla.org/show_bug.cgi?id=1329252

diff --unified --new-file --recursive --show-c-function torbrowser-45.6.0_p608/build/autoconf/icu.m4 torbrowser-45.6.0_p608-fix/build/autoconf/icu.m4
--- torbrowser-45.6.0_p608/build/autoconf/icu.m4	2017-01-15 12:04:11.207854880 +0100
+++ torbrowser-45.6.0_p608-fix/build/autoconf/icu.m4	2017-01-15 12:06:34.532291111 +0100
@@ -73,7 +73,7 @@ if test -n "$USE_ICU"; then
         fi
     fi
 
-    version=`sed -n 's/^[[:space:]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
+    version=`sed -n 's/^[[[:space:]]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
     if test x"$version" = x; then
        AC_MSG_ERROR([cannot determine icu version number from uvernum.h header file $lineno])
     fi
